Assets/Scripts/DrawTrajectory.cs

```diff
@@ -55,7 +56,7 @@ void Update()
     {
         timeSinceLastUpdateSeconds += Time.deltaTime;
 
-        if (timeSinceLastUpdateSeconds > collisionUpdateRateSeconds)
+        if (timeSinceLastUpdateSeconds > collisionUpdateRateSeconds && subject.GetComponent<KatamariMovement>().isGolfHitMode)
```

Comment:
We already have a reference to the `KatamariMovement` in this script, so it should be a little faster and simpler to use that
```suggestion
        if (timeSinceLastUpdateSeconds > collisionUpdateRateSeconds && km.IsGolfHitMode()
```